presto-main/src/main/java/com/facebook/presto/execution/SqlQueryManager.java

```diff
@@ -306,6 +328,14 @@ public QueryInfo createQuery(Session session, String query)
             // This is intentionally not a method, since after the state change listener is registered
             // it's not safe to do any of this, and we had bugs before where people reused this code in a method
             URI self = locationFactory.createQueryLocation(queryId);
+
+            // if session creation failed, create a minimal session object
+            if (session == null) {
+                session = Session.builder(new SessionPropertyManager())
+                        .setQueryId(queryId)
+                        .setIdentity(sessionFactory.getIdentity())
```

Comment:
[Issue]
AFAIU This seems like a potential problem with security. It's possible that creating session will fail in access control because of real permissions problem for that identity. And here we will create Session with "validated" user identity.